firmware/vcv_plugin/internal/make_element.cc

```diff
@@ -214,11 +217,9 @@ static Element make_slideswitch(rack::app::SvgSlider *widget) {
 		element.num_pos = std::clamp<size_t>(element.num_pos, 2, element.pos_names.size());
 	}
 
-	// Check if pq is actually a SwitchQuantity with labels
-	auto switchPq = dynamic_cast<rack::engine::SwitchQuantity*>(pq);
-	if (switchPq) {
-		for (auto i = 0u; i < std::min<size_t>(element.num_pos, switchPq->labels.size()); i++) {
-			element.pos_names[i] = switchPq->labels[i];
+	if (pq) {
+		for (auto i = 0u; i < std::min<size_t>(element.num_pos, pq->labels.size()); i++) {
```

Comment:
See above comment: do something like this (`std::min`)to ensure we don't access pq->labels out of range.